src/modules/flow/iio/nodes.c

```diff
@@ -25,15 +25,39 @@
 
 #include <sol-iio.h>
 
-struct gyroscope_data {
+enum iiodevice_types {
+    IIO_GYROSCOPE=1,
+    IIO_MAGNETOMETER=2,
+    IIO_THERMOMETER=3,
+    IIO_PRESSURE_SENSOR=4,
+    IIO_COLOR=5,
+    IIO_ACCELEROMETER=6,
+    IIO_HUMIDITY_SENSOR=7,
+    IIO_ADC=8,
+    IIO_LIGHT_SENSOR=9,
+IIO_PROXIMITY=10,
+};
+
+struct iiodevice_out {
+    union {
+        struct sol_direction_vector direction;
+        struct sol_drange drange;
+        struct sol_rgb rgb;
+    };
+};
+
+#define MAX_CHANNEL			(8)
+
+struct iiodevice_data {
     struct sol_iio_config config;
     struct sol_direction_vector scale;
     struct sol_direction_vector offset;
     struct sol_drange_spec out_range;
     struct sol_iio_device *device;
-    struct sol_iio_channel *channel_x;
-    struct sol_iio_channel *channel_y;
-    struct sol_iio_channel *channel_z;
+    struct sol_iio_channel *channels[MAX_CHANNEL];
+    enum iiodevice_types type;
+    uint16_t src_port;
```

Comment:
`src_port` is a bit misleading, don't you think? This is actually an output port. Nevertheless, you may avoid having this per node instance, if you use node type data. See flow/update or flow/ipm to see samples where this is used. Look for 'node_type' entry on their JSON. Should you have any doubt, please don't hesitate in asking.